Connections: Add redirect notice for datasources and plugins pages

mikkancso opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

7. If your PR content should be added to the What's New document for the next major or minor release, add the **add to what's new** label to your PR. Note that you should add this label to the main PR that introduces the feature; do not add this label to smaller PRs for the feature.

-->

**What is this feature?**

Add a redirect banner to the Plugins page when "All" or "Data sources" filter is set and to the Data sources page. The "Connections" button redirects to the corresponding page depending on the origin page.

[Screencast from 2023-01-05 13-18-45.webm](https://user-images.githubusercontent.com/13637610/210778919-898aca5b-833e-4ee6-9be5-8e55a1df23ec.webm)

**UPDATE:** after [this commit](https://github.com/grafana/grafana/pull/61037/commits/244399730ee5c6d0f61522c92b7b5acb4d50de71) the banner looks like this:
![Screenshot from 2023-01-05 15-06-34](https://user-images.githubusercontent.com/13637610/210799018-b9d19c47-163d-4eab-979a-a5d81062c867.png)


**Which issue(s) does this PR fix?**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes https://github.com/grafana/cloud-onboarding/issues/2886"

-->

Fixes https://github.com/grafana/cloud-onboarding/issues/2886

mikkancso committed:
add redirect notice for datasources and plugins pages

anaivanov reviewed (approved):
LGTM

leventebalogh reviewed (approved):
Nice one! 🙌

jarben commented on public/app/features/connections/components/ConnectionsRedirectNotice/ConnectionsRedirectNotice.tsx:
```diff
@@ -0,0 +1,54 @@
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  alert: css`
+  > div:nth-child(2) {
```
Just wondering whether we want to rely on the internal implementation of the Alert component, what if we keep the text as title and set something like .alertButton {position: absolute; right: 10px} ?

mikkancso replied:
I fixed this, but then we're pretty much unable to make it look like it's designed. We can still get away with it, I think.

jarben replied:
I think you can add text as a title and position the button absolutely to make it look like before?

mikkancso replied:
I'm not sure, but that also feels like a hack to me :/

mikkancso committed:
do not depend on the internal implementation of Alert

The deleted css piece was problematic because it relied on the internals
of Alert.
If we don't remove that padding, then the Alert looks a bit off, so I
added some title to the Alert to look better.

This way the Connections LinkButton is not vertically aligned to the
center, but this is the closest we can get to the designed alert without
doing hacks and reimplementing the Alert component.

jarben reviewed (approved):
LGTM
